src/core/util/fileextension.cpp

```diff
@@ -98,8 +98,12 @@ bool FileExtension::matches(std::string_view str) const {
     if (extension_.empty()) {
         return true;  // wildcard '*' matches everything
     }
-
-    return iCaseCmp(filesystem::getFileExtension(str), extension_);
+    if (str.size() >= extension_.size()) {
+        // Compare last part of path with the extension
+        return iCaseCmp(str.substr(str.size() - extension_.size()), extension_);
```

Comment:
Extension does not contain "." (same as before) unless there is a dot in the extension (nii.gz).

>  will this match abcivw the the extension ivw?

It will. I can add that as a caveat. because we never included the dot it is not possible (without major changes) to separate these. I don't see this as an issue as basically all files have an extension

Reply:
Buy the old match required the dot right? You can just check that char...

Reply:
No, the old one is the same

Reply:
How? Get file extension will require there to be a dot or else it will be empty, hence no match, what am I missing?

Reply:
One could say that the answer is yes and no. The function is used in two different ways. The first case is getting all readers by passing in only the extension (e.g. png) without knowledge about the actual file. The second case has access to the file path and here we extracted the extension before. I removed the extension extraction part.

Come to think of it, a solution would be to split into two functions. getReadersForFile and getReadersForExtension

Reply:
Ahh ic, yes that seems reasonable